pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -66,27 +66,33 @@ package {{ if and .MultiGroup .Resource.Group }}{{ .Resource.PackageName }}{{ el
 import (
```

Comment:
@NikhilSharmaWe,

 
The new occurs because the recorder is empty and needs to be passed when we create the controller and pass it in the manager ( main.go ):

```
if err = (&controllers.FoorxobReconciler{
   Client:   mgr.GetClient(),
   Scheme:   mgr.GetScheme(),
   Recorder: mgr.GetEventRecorderFor("mykind-controller"),
}).SetupWithManager(mgr); err != nil {
   setupLog.Error(err, "unable to create controller", "controller", "Foorxob")
   os.Exit(1)
}
//+kubebuilder:scaffold:builder
```


Also, in the controller, it needs to be changed to be Upper case: Recorder record.EventRecorder
Then, when we call the event use Eventf instead of Event  since we want to format the message

See that this PR does not have the e2e check as well. It is missing can you come back that?